homeassistant/components/airnow/__init__.py

```diff
@@ -47,7 +47,16 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     api_key = entry.data[CONF_API_KEY]
     latitude = entry.data[CONF_LATITUDE]
     longitude = entry.data[CONF_LONGITUDE]
-    distance = entry.data[CONF_RADIUS]
+
+    # Initialize the options parameter if the integration was first set up
+    # before OptionsFlow was added
+    if not entry.options:
+        hass.config_entries.async_update_entry(
+            entry, options={CONF_RADIUS: entry.data[CONF_RADIUS]}
```

Comment:
We should also remove radius from entry.data here to not store twice

Reply:
How is best to delete a key from `entry.data`... it is a `MappingProxyType` which is supposed to be read-only, so `pop()` and `del` don't pass checks.  I'm not sure it's worth doing a lot to remove the extra key here since this only gets invoked if the integration was set up prior to this PR to add an options flow.

Reply:
Create a new dict with the remaining settings in `data` and then in the same call to `async_update_entry()` update it.
```python
new_data = entry.data.copy()
del new_data[CONF_RADIUS]
hass.config_entries.async_update_entry(
            entry, data=new_data, options={CONF_RADIUS: entry.data[CONF_RADIUS]}
)
```